### ISSUE TITLE
Move finance tracker into CapX

### ISSUE THREAD

**Issue by aharwood2** [UoMResearchIT/CapX#269]
We currently have a invoice tracking spreadsheet. This can easily be moved into CapX so that we can automatically track funds received. Konjera can have a CapX account to view this information too with a new role created. Easy enough to migrate the existing data in.

- [x] Create DB table to store invoices.
- [x] Create service.
- [x] Migration to read in all the existing invoices from the tracker and decommission tracker.
- [x] Migration to add a "payment received" entry for those projects where there was no invoice but salary costs have been received
- [x] Drop Funds Received column from Project and make it a dynamic value created by summing all payments + edit the add/edit project form
- [x] Create data grid + add/edit/delete forms with validation to manually add and edit payments and invoices
- [x] Auto post a comment tagged with finance info when an entry is changed -- new invoice record added, or status changed. Will need to swap the "Role" to "User" and swap over with a migration the associated between Notes and People to be Note and User so users without a person entity attached can post notes.
- [x] Add line to the finance graph to show amount in outstanding invoices as well as an entry on the project summary.
- [x] Add finance role and give Konjera an account with guidance on what is expected - depends on #441 being solved.
- [x] Export button on the invoice page to export the data grid as an excel sheet.

**Comment by aharwood2** [comment]
This is on hold while we get some idea of what the new RF recharge process will be. All this work could be for nothing if they say "no invoices".

**Comment by aharwood2** [comment]
The game has changed again with RF holding out on doing anything and Rob et al. driving the TRAC route hard. It is now looking likely again that we will end up with an invoice-based approach again so this is back on the table.
